src/main/java/com/mixpanel/android/util/RemoteService.java

```diff
@@ -10,7 +10,7 @@
 
 
 public interface RemoteService {
-    boolean isOnline(Context context);
+    boolean isOnline(Context context, OfflineMode offlineMode);
```

Comment:
On the other hand, remember that if a client can provide his own implementation of `RemoteService`, then you lose the ability to manage the communication with backend services on the most efficient way for both ends (think on bad client implementations that retry every 100ms for 1 hour). 

Also, currently `RemoteService` has 3 responsabilities:
1.  check if mixpanel is blocked
2. determine when `RemoteService` is online 
3. perform mixpanel request and return the response

It is a lot to take care as a client if the only thing that I want to decide is when to allow network access to the API. As a suggestion I would change `Remote`the interface to: 

```
public interface RemoteService {
    boolean isOnline();
    byte[] performRequest(String endpointUrl, Map<String, Object> params) throws Exception;
}
```
if `Context`, `SSLSocketFactory` and `OfflineMode` are dependencies, they should be required for the implementor avoiding enforcing this on the api.